File: 2016/3/main.py
def validate_triangle(a, b, c):
    return all([(a + b > c), (b + c > a), (c + a > b)])


assert validate_triangle(5, 10, 25) == False

valid_triangles = 0
with open("input.txt") as f:
    for line in f.readlines():
        a, b, c = line.split()
        if validate_triangle(int(a), int(b), int(c)):
            valid_triangles += 1

print(valid_triangles)

File: 2016/3/main.hs
import Data.List (transpose)
import Data.List.Split (splitOn)

testTriangle :: [Int] -> Bool
testTriangle [a, b, c] = a + b > c && b + c > a && c + a > b
testTriangle _ = False

arrange :: [[Int]] -> [[Int]]
arrange [] = []
arrange sides = transpose (take 3 sides) ++ arrange (drop 3 sides)

main :: IO ()
main = do
  lines <- lines <$> readFile "input.txt"
  let sides = map read . filter (/= "") . splitOn " " <$> lines :: [[Int]]
  print $ length $ filter testTriangle sides
  print $ length $ filter testTriangle $ arrange sides
